python/cudaq/kernel/ast_bridge.py

```diff
@@ -1434,6 +1434,52 @@ def visit_Attribute(self, node):
                     self.pushValue(self.getConstantInt(hash(channel_class)))
                     return
 
+        # Handle attribute access on call results (e.g., M(6, 9).i)
```

Comment:
I think this looks good.

Q: What about a path of properties?  e.g. `foo().a.bar().b.c.baz()`?

Reply:
I think it is ok to postpone until we can test nested data classes.

Reply:
First, it was a question and I was expecting the answer to be "no". Thanks for confirming.

Second, it should be a general aim to design and build code for future possibilities/eventualities. That's just simply better design (open design) than the alternative of thinking only about all the current limitations and restrictions and hoping to ignore everything else as long as possible (box canyon design).

Finally, we obviously do not need to support this immediately nor in this PR. But, we definitely should be thinking of these language features and how to design the bridge to better handle them.